src/script/interpreter.cpp

```diff
@@ -1827,6 +1837,118 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_PUSHCURRENTINPUTINDEX:
+                {
+                    // OP_PUSHCURRENTINPUTINDEX is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    if (checker.GetnIn() > 0x7fffffff)
```

Comment:
We can probably make this an assert.  Having more that MAX_SIZE (2^25) inputs is illegal.

Reply:
Apparently In BaseSignatureChecker this is set to 0xFFFFFFFF in exotic contexts, so I guess we cannot assume anything.  I guess we can keep it as is.

Reply:
Maybe add a comment.

Reply:
I'm not a big fan of the use of this sentinel values.  the 0x7ffff is just too arbitrary.  There are a variety of things that I think would be better.

1. Testing against 0xffffff and assigning it to a static constant in the BaseSignatureChecker class.
2. Comparing with MAX_SIZE instead of 0x7ffffff here.
3. Having GetnIn return a const pointer to an unsigned int and return nullptr in BaseSignatureChecker.
4. Having GetnInt return an Optional integer.

Reply:
5. have GetnIn return an int or a int32_t and check if the value is negative.

I think this one is my favourite.

Reply:
I think checking with MAX_SIZE makes some sense. It would still require casting `unsigned int nIn` to some `int` type.